### PR TITLE
Update unit test and golden numbers in KPL

### PR DESCRIPTION
**Pull request by qlzh727**
# What does this PR do?
The those numbers are actually affected by the global RNG seeds, which might change when
we try to enable tf.random.Generator on keras side.

Set the seed to the testing layer actually fix this issue.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

### CODE REVIEW

**Review by tanzhenyu — APPROVED** [review]

**Review by LukeWood — APPROVED** [review]

**Comment by qlzh727** [comment]
Seems that the GCB is failing due to resource issue. The github action is passing and covered this unit test. Merging now.

**Comment by LukeWood** [comment]
yeah, GCB can only run for 1 pr at once
